pkg/kubeapiserver/options/authentication_test.go

```diff
@@ -261,10 +262,21 @@ func TestAuthenticationValidate(t *testing.T) {
 			options.ServiceAccounts = testcase.testSA
 			options.WebHook = testcase.testWebHook
 			options.AuthenticationConfigFile = testcase.testAuthenticationConfigFile
+
+			// SetFeatureGateEmulationVersionDuringTest needs to be called
+			// before any calls to SetFeatureGateDuringTest to work reliably.
+			for _, f := range testcase.disabledFeatures {
```

Comment:
nit: since we can only emulate a single version per test, it might be clearer to make `emulationVersion *version.Version` part of the testcase struct, explicitly set it to `version.MustParse("1.33")` in the cases that want to disable AnonymousAuthConfigurableEndpoints (with a comment why), then make this logic just do:

```go
if testcases.emulationVersion != nil {
  featuregatetesting.SetFeatureGateEmulationVersionDuringTest(t, utilfeature.DefaultFeatureGate, testcase.emulationVersion)
}
```